spec.html

```diff
@@ -36563,27 +36563,137 @@
+        <p>A <dfn>block Environment Record</dfn> is a declarative Environment Record that is used to represent block scopes. Functions declared inside the block are stored in a separate Environment Record in the [[FunctionsInBlock]] internal slot in addition to in the block environment record. This helps ensure that only function Objects are assigned to outer environments for web legacy compatibility.</p>
+        <p>Block Environment Records have the additional state field listed in <emu-xref href="#table-block-environment-record-additional-fields"></emu-xref>.</p>
+        <emu-table id="table-block-environment-record-additional-fields" caption="Additional Fields of Block Environment Records">
+          <table>
+            <tbody>
+              <tr>
+                <th>
+                  Field Name
+                </th>
+                <th>
+                  Value
+                </th>
+                <th>
+                  Meaning
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  [[FunctionsInBlock]]
+                </td>
+                <td>
+                  Declarative Environment Record
+                </td>
+                <td>
+                  Functions declared inside the block.
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </emu-table>
+        <p>Block Environment Records support all of the declarative Environment Record methods listed in <emu-xref href="#table-15"></emu-xref> and have no additional methods.</p>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-newblockenvironment">
+        <h1>NewBlockEnvironment (_E_)</h1>
+        <p>When the abstract operation NewBlockEnvironment is called with a Lexical Environment as argument _E_ the following steps are performed:</p>
+        <emu-alg>
+          1. Let _env_ be a new Lexical Environment.
+          1. Let _envRec_ be a new block Environment Record containing no bindings.
+          1. Let _fibRec_ be a new declarative Environment Record containing no bindings.
+          1. Set _envRec_.[[FunctionsInBlock]] to _fibRec_.
+          1. Set _env_'s EnvironmentRecord to be _envRec_.
+          1. Set the outer lexical environment reference of _env_ to _E_.
+          1. Return _env_.
+        </emu-alg>
+      </emu-annex>
+      <emu-annex id="sec-web-compat-block-static-semantics-early-errors">
+        <h1>Changes to Block Early Errors</h1>
+        <p>In Static Semantics: Early Errors of <emu-grammar>Block : `{` StatementList `}`</emu-grammar> (<emu-xref href="#sec-block-static-semantics-early-errors"></emu-xref>) the following replaces the first list item:</p>
+        <ul>
+          <li>
+            It is a Syntax Error if the LexicallyDeclaredNames of StatementList contains any duplicate entries, unless the source code matching this production is not strict mode code and the duplicate entries are only bound by FunctionDeclarations.
```

Comment:
StatementList -> |StatementList|
FunctionDeclarations -> |FunctionDeclaration|s

And the early errors for the [`switch` statement](http://tc39.github.io/ecma262/#sec-switch-statement-static-semantics-early-errors) need to modified in Annex B, too.